distributed/scheduler.py

```diff
@@ -2097,7 +2097,7 @@ def decide_worker_rootish_queuing_disabled(
         else:
             # Last-used worker is full, unknown, retiring, or paused;
             # pick a new worker for the next few tasks
-            ws = min(pool, key=partial(self.worker_objective, ts))
+            ws = min(pool, key=self.worker_objective_no_deps)
```

Comment:
Queued tasks would now consider occupancy too (as well as memory). This is fine, since the definition of `idle` will still prevent oversaturation. If root tasks had different duration estimates, we'd _maybe_ assign to workers a little less evenly: workers with quick tasks would fully fill before workers with slow tasks. But _all workers would be fully utilized regardless_.